src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1208,10 +1208,88 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             LowerHWIntrinsicFusedMultiplyAddScalar(node);
             break;
 
+        case NI_Sve_ConditionalSelect:
+        {
+            GenTree* op1 = node->Op(1);
+            if (op1->OperIsHWIntrinsic(NI_Sve_ConvertVectorToMask))
```

Comment:
Why would this optimization only apply to `ConvertVectorToMask`? I believe @BruceForstall added some support for TYP_MASK constants and we may want to check for them as well.

Reply:
I am not sure which code path other than `ConvertVectorToMask` would hit this. I was planning to add more scenarios once we see the need. For SVE context, we would always have `ConvertVectorToMask` as first operand for ConditionalSelect AFAIK.

Reply:
not needed anymore as I moved this handling in importer.

Reply:
Anything that directly has a `TYP_MASK`.

For example, if you did `Sve.ConditionalSelect(Sve.CompareGreaterThan(x, y), ..., ...)` then `Sve.CompareGreaterThan` would produce a `TYP_MASK` and wrap that in a `MaskToVector`. The `Sve.ConditionalSelect` would then wrap that in a `VectorToMask`. Morph has an optimization to adjust that to just consume the `TYP_MASK` directly.

Similarly since we have support for `TYP_MASK` constants, prior constant folding should be able to produce a `CNS_MSK` that we can check.

----------------------------

So, while what the code is doing currently is looking for a `VectorToMask(CNS_VEC)` , it more ideally would have already been converted to a `CNS_MSK` instead and we'd be simply checking that instead.

Having some kind of general `IsMaskAllBitsSet` or `IsMaskZero` helper might then make more sense, where we can trivially have it check `CNS_MSK` and optionally have it also check for cases like `ConvertVectorToMask(CNS_VEC)` where the `CNS_VEC` matches `IsVectorZero` or `IsVectorAllBitsSet`